### ISSUE TITLE
Using a Layout defined in BlazorApp.Client project results in a exception during SSR

### ISSUE THREAD

**Issue by oliverw** [dotnet/aspnetcore#51648]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Using a Layout defined in `BlazorApp.Client` (Blazor WASM) project results in a exception during SSR:

> InvalidOperationException: Cannot pass the parameter 'Body' to component 'FooLayout' with rendermode 'InteractiveAutoRenderMode'. This is because the parameter is of the delegate type 'Microsoft.AspNetCore.Components.RenderFragment', which is arbitrary code and cannot be serialized.

If this is working as intended then it should be mentioned somewhere that you cannot use client side layouts with pre-rendering.

### Expected Behavior

Layout in Client side project is rendered as @body of server side layout.

### Steps To Reproduce

1. Created a Blazor Web App project in RC2
2. Added a layout to `BlazorApp1.Client` project

```
@inherits LayoutComponentBase
@rendermode InteractiveAuto

<div>
    <h1>Foo</h1>
    <div class="col-md-9">
        @Body
    </div>
</div>
```

3. Modified counter.razor to use the layout:

```
@page "/counter"
@using BlazorApp1.Client.Components
@layout FooLayout

@rendermode InteractiveAuto

<PageTitle>Counter</PageTitle>

<h1>Counter</h1>

<p role="status">Current count: @currentCount</p>

<button class="btn btn-primary" @onclick="IncrementCount">Click me</button>

@code {
    private int currentCount = 0;

    private void IncrementCount()
    {
        currentCount++;
    }
}
```

Added `.AddInteractiveServerComponents();` and `.AddInteractiveServerRenderMode()` to Program.cs of Server project.

### .NET Version

8.0.100-rc.2.23502.2

**Comment by javiercn** [comment]
@oliverw thanks for contacting us.

The render mode can't be in the layout, as the parameters for the top-level components that render interactively need to be serializable, and the layout takes in a RenderFragment, which is non serializable.

**Comment by SteveSandersonMS** [comment]
You need to remove `@rendermode InteractiveAuto` from the layout file, and instead put the rendermode on the other pages or components where you want it.

**Comment by SteveSandersonMS** [comment]
Also, putting an interactive rendermode on the layout suggest you want everything to be interactive globally. If that's the case, there's a simpler way to define this - see the new project templates in RC2 that have the "global interactivity" option.

**Comment by oliverw** [comment]
> You need to remove `@rendermode InteractiveAuto` from the layout file, and instead put the rendermode on the other pages or components where you want it.

Yes, of course. Apologies for that question. What I actually wanted to ask was how do I get the client side layout render as @Body of the server side layout.

**Comment by SteveSandersonMS** [comment]
Check the RC2 project template and its "global interactivity" option. If that doesn't do what you're looking for, please let us know (and with more info about in what way it's not what you want).
